### PR TITLE
SDK-47 -- Add new standard events

### PR DESCRIPTION
**Pull request by apeterson-branch**
## Reference
SDK-47 -- Add new standard events to the Android SDK
DEVEX-876 -- Add new standard events to the Android SDK

## Description
We need to add 7 new standard events to the Android SDK. Some of these allow the passing of metadata, via the form of a BUO, some do not. The last four are Facebook events, which have a different type of metadata.

Acceptance Criteria: As an developer integrating the Branch SDK, I can call one of these new 7 events, anytime a user of my app hits these actions, so that I can measure and collect event traffic through the dashboard and data feeds.

## Testing Instructions
As part of this release, I added a basic unit test framework.  While this will eventually be filled out, I only added tests necessary to this PR.

* Sample app should continue to function.
* Unit tests should all pass

### NOTES
* I was unable to add unit tests that could prove redaction of BUO on the subset of events that required this per the PRD.  This will need to be tested externally.

## Risk Assessment [`MEDIUM`]
This is a more complicated issue than just adding a couple of new events, as these have some side effects that wind through the code.

- [x] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [ ] JIRA Ticket is referenced in MR title
- Correctness & Style
    - [ ] Conforms to [Style Guides](https://google.github.io/styleguide/javaguide.html)
    - [ ] Mission critical pieces are documented in code and out of code as needed
- [ ] Unit Tests reviewed and test issue sufficiently
- [ ] Functionality was reviewed in QA independently by another engineer on the team (clone for each required reviewer)

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/util/BranchEvent.java [hunk 1]
```diff
@@ -44,13 +45,6 @@ private BranchEvent(String eventName, boolean isStandardEvent) {
         customProperties = new JSONObject();
         this.eventName = eventName;
 
-        for (BRANCH_STANDARD_EVENT event : BRANCH_STANDARD_EVENT.values()) {
```
[the remark this answers is not in the document]
Comment (aaaronlopez) [reply Branch-SDK/src/io/branch/referral/util/BranchEvent.java]: I literally had this comment open in the review over a month ago, and I just realized I hadn't submitted...

File: Branch-SDK/src/io/branch/referral/util/BranchEvent.java [hunk 2]
```diff
@@ -280,6 +377,33 @@ protected boolean shouldUpdateLimitFacebookTracking() {
         public boolean shouldRetryOnFail() {
             return true; // Branch event need to be retried on failure.
         }
+
+        /**
+         * Some V2 Standard Events need to have their BUO redacted.
+         * @return {code true} if the Event BUO should be redacted.
+         */
+        private boolean redactBuo() {
```
Comment (aaaronlopez) [inline Branch-SDK/src/io/branch/referral/util/BranchEvent.java]: This is not really your thing, but this whole BUO redaction kinda makes me scratch my head. My question as a layperson would be, "Why let me attach a BUO in the first place then?" "Why doesn't it show up on my dashboard if I attached it?"

Comment (apeterson-branch) [reply Branch-SDK/src/io/branch/referral/util/BranchEvent.java]: This is a great question.  The PRD was very specific on this though, and a review of that document should be done to make sure that we are progressing in the right direction.

**Comment by apeterson-branch** [comment]
Closing this PR as obsolete.

**Pull request closed without merging**